server/notifications/user_notifier_test.go

```diff
@@ -7,47 +7,83 @@ import (
 	"github.com/Hukyl/genesis-kma-school-entry/models"
 	"github.com/Hukyl/genesis-kma-school-entry/rate"
 	"github.com/Hukyl/genesis-kma-school-entry/server/notifications"
-	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
 )
 
 type mockRateFetcher struct {
-	callCount int
+	mock.Mock
 }
 
 func (m *mockRateFetcher) FetchRate(from, to string) (rate.Rate, error) {
-	m.callCount++
-	return rate.Rate{CurrencyFrom: from, CurrencyTo: to, Rate: 27.5}, nil
+	args := m.Called(from, to)
+	return args.Get(0).(rate.Rate), args.Error(1)
 }
 
 type mockUserRepository struct {
-	callCount int
+	mock.Mock
 }
 
 func (m *mockUserRepository) FindAll() ([]models.User, error) {
-	m.callCount++
-	return []models.User{
-		{Email: "example@gmail.com"},
-		{Email: "example2@gmail.com"},
-	}, nil
+	args := m.Called()
+	return args.Get(0).([]models.User), args.Error(1)
+}
+
+type mockMessageFormatter struct {
+	mock.Mock
+}
+
+func (m *mockMessageFormatter) SetRate(rate rate.Rate) {
+	m.Called(rate)
+}
+
+func (m *mockMessageFormatter) Subject() string {
+	args := m.Called()
+	return args.String(0)
+}
+
+func (m *mockMessageFormatter) String() string {
+	args := m.Called()
+	return args.String(0)
 }
 
 type mockEmailClient struct {
-	callCount int
+	mock.Mock
 }
 
-func (m *mockEmailClient) SendEmail(_ context.Context, _, _ string) error {
-	m.callCount++
-	return nil
+func (m *mockEmailClient) SendEmail(ctx context.Context, email, subject, message string) error {
+	args := m.Called(ctx, email, subject, message)
+	return args.Error(0)
 }
 
-func TestNotify(t *testing.T) {
+func TestUserNotify(t *testing.T) {
 	ctx := context.Background()
-	rateFetcher := &mockRateFetcher{}
-	userRepository := &mockUserRepository{}
-	emailClient := &mockEmailClient{}
-	notifier := notifications.NewUsersNotifier(emailClient, rateFetcher, userRepository)
+	rateFetcher := new(mockRateFetcher)
+	rateFetcher.On("FetchRate", "USD", "UAH").Return(rate.Rate{Rate: 27.5}, nil)
+	userRepository := new(mockUserRepository)
+	userRepository.On("FindAll").Return([]models.User{
+		{Email: "example@gmail.com"},
+		{Email: "example2@gmail.com"},
+	}, nil)
+	emailClient := new(mockEmailClient)
+	emailClient.On(
+		"SendEmail", ctx, "example@gmail.com", mock.Anything, mock.Anything,
+	).Return(nil).Once()
+	emailClient.On(
+		"SendEmail", ctx, "example2@gmail.com", mock.Anything, mock.Anything,
+	).Return(nil).Once()
+	messageFormatter := new(mockMessageFormatter)
```

Comment:
can you add blank line after each mock initialization, please? cus it's a bit hard to read